v0/destinations/wootric/transform.js

```diff
@@ -0,0 +1,190 @@
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  getIntegrationsObj,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  formatTimeStamp,
+  CustomError
+} = require("../../util");
+const { getAccessToken, retrieveUserId } = require("./util");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const { set } = require("lodash");
+
+const responseBuilder = async (payload, endpoint, method, destination) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    const accessToken = await getAccessToken(destination);
+    if (!accessToken) {
+      throw new CustomError(`[Wootric]:: access token is not available`, 400);
+    }
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Bearer ${accessToken}`,
+      Accept: "application/json"
+    };
+    response.method = method;
+    response.body.FORM = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new CustomError("Payload could not be constructed", 400);
+};
+
+const identifyResponseBuilder = async (message, destination) => {
+  let payload;
+  let endpoint;
+  let method;
+
+  const userId =
+    message.userId ||
+    message.traits?.userId ||
+    message.traits?.id ||
+    message.context?.traits?.userId ||
+    message.context?.traits?.id ||
+    message.anonymousId;
+
+  const wootricEndUserId = await retrieveUserId(userId, destination);
+  if (!wootricEndUserId) {
+    payload = constructPayload(
+      message,
+      MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_USER.name]
+    );
+    endpoint = CONFIG_CATEGORIES.CREATE_USER.endpoint;
+    method = "POST";
+
+    if (!payload.properties.email && !payload.properties.phone_number) {
+      throw new CustomError(
+        "email/phone number are missing. At least one parameter must be provided",
+        400
+      );
+    }
+  } else {
+    payload = constructPayload(
+      message,
+      MAPPING_CONFIG[CONFIG_CATEGORIES.UPDATE_USER.name]
+    );
+    endpoint = CONFIG_CATEGORIES.UPDATE_USER.endpoint.replace(
+      "<end_user_id>",
+      wootricEndUserId
+    );
+    method = "PUT";
+  }
```

Comment:
can we think about lowering redunduncy by moving all of these into a util function? We can pass the changing value as parameters. WDYT?

Reply:
we can move the redundant code in util